app/assets/javascripts/Components/Modals/filter_modal.jsx

```diff
@@ -0,0 +1,317 @@
+      annotationValue: this.state.currentAnnotationValue,
+      sectionValue: this.state.currentSectionValue,
+      markingStateValue: this.state.currentMarkingStateValue,
+      tas: this.state.currentTas,
+      tags: this.state.currentTags,
+      totalMarkRange: this.state.currentTotalMarkRange,
+      totalExtraMarkRange: this.state.currentTotalExtraMarkRange,
+    });
+    this.props.onRequestClose();
+  };
+
+  clearFilters = event => {
+    event.preventDefault();
+    this.setState(INITIAL_MODAL_STATE);
+  };
+
+  render() {
+    if (this.props.loading) {
+      return "";
+    }
+    return (
+      <div>
+        <Modal
+          className="react-modal dialog"
+          isOpen={this.props.isOpen}
+          onRequestClose={() => {
+            this.setState({
+              currentOrderBy: this.props.filterData.orderBy,
+              currentAscBool: this.props.filterData.ascBool,
+              currentAnnotationValue: this.props.filterData.annotationValue,
+              currentSectionValue: this.props.filterData.sectionValue,
+              currentMarkingStateValue: this.props.filterData.markingStateValue,
+              currentTas: this.props.filterData.tas,
+              currentTags: this.props.filterData.tags,
+              currentTotalMarkRange: this.props.filterData.totalMarkRange,
+              currentTotalExtraMarkRange: this.props.filterData.totalExtraMarkRange,
+            });
+            this.props.onRequestClose();
+          }}
+        >
+          <h3>{I18n.t("results.filters.filter_by")}</h3>
+          <form onSubmit={this.onSubmit}>
+            <div className={"modal-container-scrollable"}>
+              <div className={"modal-container-vertical"}>
+                <div className={"modal-container"}>
+                  <div className={"filter"} data-testid={"order-by"}>
+                    <p>{I18n.t("results.filters.order_by")} </p>
+                    <SingleSelectDropDown
+                      options={[
+                        I18n.t("results.filters.ordering.group_name"),
+                        I18n.t("submissions.commit_date"),
+                      ]}
+                      selected={this.state.currentOrderBy}
+                      select={selection => {
+                        this.setState({currentOrderBy: selection});
+                      }}
+                      defaultValue={I18n.t("results.filters.ordering.group_name")}
+                    />
+                    <div
+                      className={"order"}
+                      onChange={e => {
+                        this.setState({currentAscBool: !this.state.currentAscBool});
+                      }}
+                      data-testid={"radio-group"}
+                    >
+                      <input
+                        type="radio"
+                        checked={this.state.currentAscBool}
+                        name="order"
+                        value="Asc"
+                        onChange={() => {}}
```

Comment:
Define the appropriate callback here, not in the enclosing `div`. Be explicit about setting the `ascending` value to `true` or `false`